lib/AST/DiagnosticEngine.cpp

```diff
@@ -140,6 +143,39 @@ struct EducationalNotes {
 static constexpr EducationalNotes<LocalDiagID::NumDiags> _EducationalNotes = EducationalNotes<LocalDiagID::NumDiags>();
 static constexpr auto educationalNotes = _EducationalNotes.value;
 
+class LocalizationInput : public llvm::yaml::Input {
+  using Input::Input;
+
+  /// Override a member operator>> to get the diagnostics in correct order. 
+  template <typename T>
+  inline friend
+  typename std::enable_if<llvm::yaml::has_SequenceTraits<T>::value,
+                          LocalizationInput &>::type
+  operator>> (LocalizationInput &yin,
+              T &diagnostics) {
+    llvm::yaml::EmptyContext Ctx;
+    T Diags;
```

Comment:
This is still loading everything into a temporary buffer, is there a way to get Input to read YAML sequentially?

Reply:
I don't think so. Should I investigate further?

Reply:
Yeah, I think that would be good, there is no reason to build a class which does exactly what `Input` already does...